url79d4f7c7ed_schanq_Live_Algorithm_tgz-pJ8-algorithms_and_systems_AI_Evolutionary_GeneticAlgorithmJ8/src/expert_system/Rhythmic_Block.java

```diff
@@ -0,0 +1,805 @@
+			rhythmicValues[1] = Rhythmic_Value.QUAVER;
+			rhythmicValues[2] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[3] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			
+		}
+		else if(name == "CROTCHET_SWUNG_T2"){
+			rhythmicValues = new Rhythmic_Value [4];
+			rhythmicValues[0] = Rhythmic_Value.QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[2] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			rhythmicValues[3] = Rhythmic_Value.CROTCHET;
+			
+		}
+		else if(name == "CROTCHET_SWUNG_T3"){
+			rhythmicValues = new Rhythmic_Value [5];
+			rhythmicValues[0] = Rhythmic_Value.QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[2] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			rhythmicValues[3] = Rhythmic_Value.QUAVER;
+			rhythmicValues[4] = Rhythmic_Value.QUAVER;
+			
+		}
+		else if(name == "CROTCHET_SWUNG_T4"){
+			rhythmicValues = new Rhythmic_Value [6];
+			rhythmicValues[0] = Rhythmic_Value.QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[2] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			rhythmicValues[3] = Rhythmic_Value.QUAVER;
+			rhythmicValues[4] = Rhythmic_Value.DEMI_SEMI_QUAVER_REST;
+			rhythmicValues[5] = Rhythmic_Value.DOTTED_SEMI_QUAVER;
+			
+		}
+		else if(name == "TRIPLET_CROTCHETS_FULL"){
+			rhythmicValues = new Rhythmic_Value [3];
+			rhythmicValues[0] = Rhythmic_Value.TRIPLET_CROTCHET;
+			rhythmicValues[1] = Rhythmic_Value.TRIPLET_CROTCHET;
+			rhythmicValues[2] = Rhythmic_Value.TRIPLET_CROTCHET;
+
+		}
+		else if(name == "TRIPLET_CROTCHETS_FIRST_OFF"){
+			rhythmicValues = new Rhythmic_Value [3];
+			rhythmicValues[0] = Rhythmic_Value.TRIPLET_CROTCHET_REST;
+			rhythmicValues[1] = Rhythmic_Value.TRIPLET_CROTCHET;
+			rhythmicValues[2] = Rhythmic_Value.TRIPLET_CROTCHET;
+
+		}
+		else if(name == "TRIPLET_CROTCHETS_SECOND_OFF"){
+			rhythmicValues = new Rhythmic_Value [3];
+			rhythmicValues[0] = Rhythmic_Value.TRIPLET_CROTCHET;
+			rhythmicValues[1] = Rhythmic_Value.TRIPLET_CROTCHET_REST;
+			rhythmicValues[2] = Rhythmic_Value.TRIPLET_CROTCHET;
+			
+		}
+		else if(name == "TRIPLET_CROTCHETS_THIRD_OFF"){
+			rhythmicValues = new Rhythmic_Value [3];
+			rhythmicValues[0] = Rhythmic_Value.TRIPLET_CROTCHET;
+			rhythmicValues[1] = Rhythmic_Value.TRIPLET_CROTCHET;
+			rhythmicValues[2] = Rhythmic_Value.TRIPLET_CROTCHET_REST;
+
+		}
+		else if(name == "QUAVERS2_FIRST_ON"){
+			rhythmicValues = new Rhythmic_Value [2];
+			rhythmicValues[0] = Rhythmic_Value.QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.QUAVER_REST;
+
+		}
+		else if(name == "QUAVERS2_SECOND_ON"){
+			rhythmicValues = new Rhythmic_Value [2];
+			rhythmicValues[0] = Rhythmic_Value.QUAVER_REST;
+			rhythmicValues[1] = Rhythmic_Value.QUAVER;
+
+		}
+		else if(name == "QUAVERS2"){
+			rhythmicValues = new Rhythmic_Value [2];
+			rhythmicValues[0] = Rhythmic_Value.QUAVER;
+			rhythmicValues[1] = Rhythmic_Value.QUAVER;
+
+		}
+		else if(name == "QUAVERS4"){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.